src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -430,9 +430,12 @@ private void RemoveFromItems(IEnumerable<ISearchEntryViewModel> oldItems)
 
         private void AddToItems(IEnumerable<ISearchEntryViewModel> newItems)
         {
+            bool hasClasses = Items.FirstOrDefault() is ClassesNodeCategoryViewModel;
+
             foreach (var entry in newItems)
             {
-                var first = Items.Select((x, i) => new { x.Name, Idx = i })
+                var first = Items.Where(cat => !(cat is ClassesNodeCategoryViewModel)).
+                    Select((x, i) => new { x.Name, Idx = i })
```

Comment:
If `Items` contains 100 elements, then `new { x.Name, Idx = i }` is invoked 100 times to create 100 intermediate objects. Can we use a simple `for` loop to determine the index of next larger item?

```csharp
var nextLargerItemIndex = -1;
foreach (var item in Items)
{
    nextLargerItemIndex = ...;
}
```